js/src/components/app-input-number-control/index.js

```diff
@@ -36,6 +41,17 @@ const AppInputNumberControl = ( props ) => {
 	const numberSettings = useStoreNumberSettings( settings );
 	const numberFormat = useNumberFormat( settings );
 
+	// The `InputControl` in `@wordpress/components` doesn't take into account the context
+	// in which the `value` of a controlled component is updated via `onBlur`, resulting in
+	// the incoming `value` prop may not syncing to the <input> element's display value.
+	// Therefore, a synchronization needle is used to make the value synchronized via the
+	// onChange callback.
```

Comment:
Thanks for the changes. They look good. The only thing is that "sync Needle" naming, seems to me a kinda personal concept name? . I was checking around the internet and I didn't find anything regarding "synchronisation needle". So I would suggest using something more descriptive like `refreshInput`

Reply:
The naming was inspired by the reset needle.

![resetrouter](https://github.com/woocommerce/google-listings-and-ads/assets/17420811/a301cb0e-fc4f-47e4-8ece-efca081fb117)

Changed to adopt [an official approach and naming](https://legacy.reactjs.org/docs/hooks-faq.html#is-there-something-like-forceupdate) in https://github.com/woocommerce/google-listings-and-ads/pull/2019/commits/f763449a5b47efffe9b3ddd3a9a12dc711563a56.

Reply:
Much better, thanks for this and for the reference.